.github/workflows/main.yml

```diff
@@ -0,0 +1,165 @@
+name: Test, build and deploy
+
+on:
+  push: {}
+  pull_request: {}
+  schedule:
+    - cron: '0 0 * * *'
+
+env:
+  STREAMLINK_DIST_DIR: ${{ github.workspace }}/dist
+  SIGNING_KEY_ID: 2E390FA0
+  SIGNING_KEY_FILE: ${{ github.workspace }}/signing.key
+
+jobs:
+  test:
+    name: Run tests
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+        python: [2.7, 3.5, 3.6, 3.7, 3.8]
+#       include:
+#         - python: X.Y-dev
+#           continue: true
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install dependencies
+        continue-on-error: ${{ matrix.continue }}
+        run: bash ./script/install-dependencies.sh
+      - name: Test
+        continue-on-error: ${{ matrix.continue }}
+        run: pytest --cov
+      - name: Lint
+        continue-on-error: ${{ matrix.continue }}
+        run: flake8
+      - name: Upload coverage data
+        continue-on-error: ${{ matrix.continue }}
+        run: codecov
+
+  documentation:
+    name: Test documentation
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 300
+      - name: Fetch tags
+        run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+      - uses: actions/setup-python@v1
+        with:
+          python-version: 3.5
```

Comment:
What's the reasoning for building with 3.5 here and below instead of a more recent release?

Reply:
The same version is currently set in the travisci config https://github.com/streamlink/streamlink/blob/d4b00527aae0bea34a453b9b0e53caf56d7ac2bc/.travis.yml#L14-L15
I didn't want to introduce more unnecessary changes, but yes, you're right, we should change that.
I'll update this tomorrow in one go with some other things. It'll also be checked/verified on my test repo then.

Reply:
@bastimeyer It's not a huge deal so don't stress over it, just figured since we're updating so much it's worth a discussion to change it. I'm fine with sticking on 3.5 if you want for a bit.